modules/core/04-channel/keeper/upgrade.go

```diff
@@ -0,0 +1,92 @@
+package keeper
+
+import (
+	"reflect"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	connectiontypes "github.com/cosmos/ibc-go/v7/modules/core/03-connection/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v7/modules/core/24-host"
+)
+
+// ChanUpgradeInit is called by a module to initiate a channel upgrade handshake with
+// a module on another chain.
+func (k Keeper) ChanUpgradeInit(
+	ctx sdk.Context,
+	portID string,
+	channelID string,
+	chanCap *capabilitytypes.Capability,
+	proposedUpgradeChannel types.Channel,
+	counterpartyTimeoutHeight clienttypes.Height,
+	counterpartyTimeoutTimestamp uint64,
+) (uint64, string, error) {
```

Comment:
nit: can you name the return args here so i know the string here is supposed to be the previous version?